invocation/trigger_events.go

```diff
@@ -0,0 +1,400 @@
+	}
+
+	injectTriggerTagsToInvocation(domainName, className, uris)
+}
+
+func injectTriggerTagsForAPIGateway(payload json.RawMessage) {
+	e := triggerEvent{}
+	err := json.Unmarshal(payload, &e)
+	if err != nil {
+		return
+	}
+
+	domainName := domainNames["API"]
+	className := classNames["APIGATEWAY"]
+
+	path := ""
+	if e.Params.Header["Host"] != "" && e.Context["stage"] != "" {
+		path = e.Params.Header["Host"] + "/" + e.Context["stage"] + e.Context["resource-path"]
+	}
+
+	var operationNames = []string{path}
+
+	injectTriggerTagsToInvocation(domainName, className, operationNames)
+}
+
+func injectTriggerTagsForAPIGatewayProxy(payload json.RawMessage) {
+	e := events.APIGatewayProxyRequest{}
+	err := json.Unmarshal(payload, &e)
+	if err != nil {
+		return
+	}
+
+	domainName := domainNames["API"]
+	className := classNames["APIGATEWAY"]
+
+	path := ""
+	if e.Headers["Host"] != "" {
+		path = e.Headers["Host"] + "/" + e.RequestContext.Stage + e.Path
+	}
+	var operationNames = []string{path}
+
+	injectTriggerTagsToInvocation(domainName, className, operationNames)
+}
+
+func injectTriggerTagsForCloudWatchLogs(payload json.RawMessage) {
+	e := events.CloudwatchLogsEvent{}
+	err := json.Unmarshal(payload, &e)
+	if err != nil {
+		return
+	}
+
+	domainName := domainNames["LOG"]
+	className := classNames["CLOUDWATCHLOG"]
+
+	data, err := e.AWSLogs.Parse()
+	var operationNames []string
+	if err == nil {
+		operationNames = append(operationNames, data.LogGroup)
+	}
+	injectTriggerTagsToInvocation(domainName, className, operationNames)
+
+}
+
+func injectTriggerTagsForSchedule(payload json.RawMessage) {
+	e := events.CloudWatchEvent{}
+	err := json.Unmarshal(payload, &e)
+	if err != nil {
+		return
+	}
+
+	domainName := domainNames["SCHEDULE"]
+	className := classNames["SCHEDULE"]
+
+	scheduleNamesMap := make(map[string]struct{})
+	for _, resource := range e.Resources {
+		scheduleSlice := strings.Split(resource, "/")
+		scheduleName := scheduleSlice[len(scheduleSlice)-1]
+		scheduleNamesMap[scheduleName] = void
+	}
+	var operationNames []string
+	for k := range scheduleNamesMap {
+		operationNames = append(operationNames, k)
+	}
+
+	injectTriggerTagsToInvocation(domainName, className, operationNames)
+}
+
+func injectTriggerTagsForLambda(ctx context.Context) {
+	domainName := domainNames["API"]
+	className := classNames["LAMBDA"]
+
+	clientContext, ok := application.GetClientContext(ctx)
+	if ok {
+		operationName := clientContext.Custom[lambdaTriggerOperationName]
+		operationNames := []string{operationName}
+		injectTriggerTagsToInvocation(domainName, className, operationNames)
+	}
+}
+
+func setInvocationTriggerTags(ctx context.Context, payload json.RawMessage) {
+	ok := injectTriggerTagsFromInputType(ctx, payload)
+	if !ok {
+		injectTriggerTagsFromPayload(ctx, payload)
+	}
+}
+
+func injectTriggerTagsFromInputType(ctx context.Context, payload json.RawMessage) bool {
+	eventType, ok := ctx.Value(eventTypeKey{}).(reflect.Type)
```

Comment:
I think that we shouldn't be getting the `eventType` from the `ctx` by using the `eventTypeKey` type that we already declared in the `agent` package. There should be a common place and the necessary methods to access `eventType` information in the context using the `eventTypeKey`. That `eventTypeKey`type shouldn't be recreated each time we need it, instead it should be declared once as a package private type.